projects/hathor-wallet-mobile/0002-nano-contract-integration.md

```diff
@@ -0,0 +1,476 @@
+
+## Unleash
+[unleash]: #unleash
+
+First, we should enable the user's device in the `nano-contract.rollout` feature flag on Unleash. For this feature flag we use the default variant with 3 strategies:
+1. UserIds: this first strategy is dedicated for internal testers
+2. UserIds: this second strategy is dedicated for external testers
+3. Gradual rollout: this third strategy is dedicated to the public
+
+Each strategy [must](https://www.rfc-editor.org/rfc/rfc2119) support the following values:
+- `platform`: `ios`, `android`
+- `network`: `mainnet`, `testnet`, `privatenet`
+- `stage`: `mainnet`, `testnet`
+
+Second, we should enable the user's device in the `wallet-connect-mobile.rollout` feature flag on Unleash.
+
+## Opt-in
+
+```mermaid
+graph LR
+    A[Settings] --> B[Nano Contract Settings]
+```
+
+### Screen: Settings
+[screen-settings]: #screen-settings
+
+The "Settings" screen is divided into two sections: "General Settings" and "Dev Settings." Within "General Settings", the *Nano Contract* option is included.
+
+This option is visible only when the device is registered with Unleash. For registration and feature flag activation, see [Unleash](#unleash).
+
+Selecting the *Nano Contract* option initiates navigation to the *Nano Contract Settings* screen. This action allows users to modify their nano-contract preferences.
+
+##### Design Bill
+- *Nano Contract* option:
+	- **Placement**: Insert this option in the "General Settings" section, positioned before "Reset wallet" item.
+	
+	- **Functionality**: The option responds to touch input. A tap triggers a transition to the *Nano Contract Settings* screen.
+	
+	- **Visibility Conditions**: Display this option based on the device's registration status with Unleash.
+
+### Screen: Nano Contract Settings
+[screen-nano-contract-settings]: #screen-nano-contract-settings
+
+The *Nano Contract Settings* screen features a "Nano Contract" toggle button. Users can tap this button to activate the nano-contract feature.
+
+Upon activation, the wallet initiates a restart process, during which Nano Contract transactions for the current wallet are loaded. Users are visually informed of this process through a progress indicator and are returned to the *Nano Contract Settings* screen upon completion.
+
+>[!NOTE]
+In addition to enabling the *Nano Contract* feature, basic usage data such as feature activation frequency and duration of use may be collected for monitoring purposes. For more details on data collection and push notification support, see [Basic Data Gathering and Push Notification Support](#basic-data-gathering-push-notification-support).
```

Comment:
I don't think we should intentionally collect any kind of data for monitoring purposes

Reply:
I think there is no problem to get anonymized data with user consent. We may want to know the feature adoption statistics, specially to prioritize our efforts. Unleash is not reliable to get adoption statistics.

Reply:
I understand your concerns, they are super valid. This ins't an easy topic. I believe it would be beneficial to provoke this discussion with the team and possibly involve the community in the process. We will spend some effort in the near feature in UI and UX improvements and do it without data will really throw us in the dark. By discussing this topic we will at least discover and map our boundaries.

Reply:
I've dicussed this topic with @pedroferreira1 and we come to a conclusion that it is the realm of product development and @trondbjoroy should lead this.